File: wear/src/main/java/at/shockbytes/corey/core/WorkoutActivity.java
package at.shockbytes.corey.core;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.wearable.activity.WearableActivity;
import android.view.KeyEvent;

import at.shockbytes.corey.R;
import at.shockbytes.corey.common.core.workout.model.Workout;
import at.shockbytes.corey.fragment.WorkoutFragment;

public class WorkoutActivity extends WearableActivity {

    public interface OnWorkoutNavigationListener {

        void moveToNext();

        void moveToPrevious();

        void onEnterAmbient();

        void onUpdateAmbient();

        void onExitAmbient();

    }

    private static final String ARG_WORKOUT = "arg_workout";

    public static Intent newIntent(Context context, Workout w) {
        return new Intent(context, WorkoutActivity.class).putExtra(ARG_WORKOUT, w);
    }

    private OnWorkoutNavigationListener navigationListener;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_workout);
        setAmbientEnabled();

        Workout w = getIntent().getParcelableExtra(ARG_WORKOUT);
        WorkoutFragment workoutFragment = WorkoutFragment.newInstance(w);
        navigationListener = workoutFragment;
        getFragmentManager().beginTransaction()
                .replace(R.id.activity_workout_container, workoutFragment)
                .commit();
    }

    @Override
    public void onEnterAmbient(Bundle ambientDetails) {
        super.onEnterAmbient(ambientDetails);
        navigationListener.onEnterAmbient();
    }

    @Override
    public void onUpdateAmbient() {
        super.onUpdateAmbient();
        navigationListener.onUpdateAmbient();
    }

    @Override
    public void onExitAmbient() {
        super.onExitAmbient();
        navigationListener.onExitAmbient();
    }

    @Override
    public boolean onKeyDown(int keyCode, KeyEvent event) {

        switch (keyCode) {
            case KeyEvent.KEYCODE_NAVIGATE_NEXT:
                navigationListener.moveToNext();
                break;
            case KeyEvent.KEYCODE_NAVIGATE_PREVIOUS:
                navigationListener.moveToPrevious();
                break;
        }

        return super.onKeyDown(keyCode, event);
    }



}

File: wear/src/main/java/at/shockbytes/corey/ui/activity/WorkoutActivity.kt
package at.shockbytes.corey.ui.activity

import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.view.KeyEvent

import at.shockbytes.corey.R
import at.shockbytes.corey.common.core.workout.model.Workout
import at.shockbytes.corey.ui.fragment.WorkoutFragment

class WorkoutActivity : AppCompatActivity() {

    interface OnWorkoutNavigationListener {

        fun moveToNext()

        fun moveToPrevious()

        fun onEnterAmbient()

        fun onUpdateAmbient()

        fun onExitAmbient()
    }

    private var navigationListener: OnWorkoutNavigationListener? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_workout)

        intent.getParcelableExtra<Workout>(ARG_WORKOUT)?.let { w ->
            val workoutFragment = WorkoutFragment.newInstance(w)
            navigationListener = workoutFragment
            supportFragmentManager.beginTransaction()
                    .replace(R.id.activity_workout_container, workoutFragment)
                    .commit()
        }
    }

    override fun onKeyDown(keyCode: Int, event: KeyEvent): Boolean {

        when (keyCode) {
            KeyEvent.KEYCODE_NAVIGATE_NEXT -> navigationListener?.moveToNext()
            KeyEvent.KEYCODE_NAVIGATE_PREVIOUS -> navigationListener?.moveToPrevious()
        }
        return super.onKeyDown(keyCode, event)
    }

    companion object {

        private const val ARG_WORKOUT = "arg_workout"

        fun newIntent(context: Context, w: Workout): Intent {
            return Intent(context, WorkoutActivity::class.java).putExtra(ARG_WORKOUT, w)
        }
    }
}
